torch_xla/csrc/index_op_util.cpp

```diff
@@ -0,0 +1,104 @@
+#include "torch_xla/csrc/index_op_util.h"
```

Comment:
The main reason for this is avoiding weird dependencies. We'd have something under `ops` depend on `at::Tensor` and `XLATensor`. Would you rather have that instead of the current situation? That was the main reason I split them.

Reply:
There is no weird dependency AFAICT.
The index_op.cpp/h contains all the infrstructure to do this thing, takes input ir::Value/Node and produces IR Node.

Reply:
There is because of the canonicalization of the null bits, if we put everything in one place.

Reply:
Done.